chapter2/tests/t19/terraform/_modules/nginx/main.tf

```diff
@@ -0,0 +1,19 @@
+module "container" {
+  cont                      = "nginx"
+  source                    = "../container"
+  container_image           = var.container_image
+  container_image_keep_locally = var.container_image_keep_locally
+  container_name            = var.container_name != null ? var.container_name : ""
+  container_ports           = var.container_ports
+  client                    = var.client
+  project                   = var.project
+  environment               = var.environment
+  container_volume_hostpath = "${abspath(path.root)}/../../${var.environment}"
+  container_volume_path     = "/usr/share/nginx/html"
+}
+
+resource "null_resource" "index_page" {
+  provisioner "local-exec" {
+    command = "mkdir ../../${var.environment} && cat > ../../${var.environment}/index.html  <<EOL\n${local.rendered_index_html}\nEOL"
```

Comment:
ise path.root or path.module to avoid `...` etc.

Reply:
this file should contain the destroy provisioner as well. The destroy should not be in the generic `container` module.

Each module should know what to destroy. Not the generic one.

Reply:
i.e. if you created something on line 17 - then you should have a destroy as well in this module. Things (logical) should be grouped together. If you have a lot of terraform code - how do I know that you created thing in module X but you deleted the same thing in module Y - that is bad design and hard to maintain - will lead to errors and chaos down the road.

Reply:
> ise path.root or path.module to avoid `...` etc.

@dmitry-mightydevops
Done

Reply:
> this file should contain the destroy provisioner as well. The destroy should not be in the generic `container` module.
> 
> Each module should know what to destroy. Not the generic one.

@dmitry-mightydevops
Done

Reply:
> i.e. if you created something on line 17 - then you should have a destroy as well in this module. Things (logical) should be grouped together. If you have a lot of terraform code - how do I know that you created thing in module X but you deleted the same thing in module Y - that is bad design and hard to maintain - will lead to errors and chaos down the road.

@dmitry-mightydevops
Done